.github/workflows/reusable-create-environment.yml

```diff
@@ -48,15 +48,32 @@ jobs:
         uses: ./.github/actions/setup-environment
         with:
           environment-name: ${{ inputs.environment-name }}
-      - name: Contentful Checks
-        id: determine-contentful-environment
+      - name: Determine whether we need to create a Contentful environment
+        id: get-contentful-state
         uses: ./.github/actions/contentful-checks
         with:
           pr-number: ${{ steps.setup.outputs.pr-number }}
           contentful-space-id: ${{ steps.setup.outputs.contentful-space-id }}
           contentful-master-env: ${{ steps.setup.outputs.contentful-environment }}
           contentful-token: ${{ secrets.CONTENTFUL_MANAGEMENT_TOKEN }}
           max-contentful-envs: ${{ steps.setup.outputs.max-contentful-envs }}
+      - name: Ensure that we are not at Contentful environment limit
+        id: ensure-limit-not-reached
+        if: ${{ steps.get-contentful-state.outputs.on-branch-env == 'True' && steps.get-contentful-state.outputs.contentful-env-exists == 'False' && steps.get-contentful-state.outputs.contentful-limit-reached == 'True' }}
+        shell: bash
+        run: |
+          echo "Contentful Environment required but all available slots are in use."
+          exit 1
+      - name: Create Contentful environment
+        id: create-contentful-environment
+        if: ${{ steps.get-contentful-state.outputs.on-branch-env == 'True' && steps.get-contentful-state.outputs.contentful-env-exists == 'False' && steps.get-contentful-state.outputs.contentful-limit-reached == 'False' }}
+        shell: bash
+        run: bash .github/scripts/contentful/create.sh
```

Comment:
do we need a script for it? why not just do it inline?

Reply:
We were creating these as actions, but we felt a script was better. It takes the functionality away from the framework of the workflows.

Also scripts give you syntax highlighting/support.

Reply:
> We were creating these as actions, but we felt a script was better. It takes the functionality away from the framework of the workflows.
> 
> Also scripts give you syntax highlighting/support.

Easier to test locally as well.